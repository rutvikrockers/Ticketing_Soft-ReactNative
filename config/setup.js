// create our store for testing redux components
import configureMockStore from 'redux-mock-store';
const thunk = require('redux-thunk');

// Setup adapter to work with enzyme 3.2.0
const Enzyme = require('enzyme');
const Adapter = require('enzyme-adapter-react-16');

export const mockStore = configureMockStore([thunk.default]);

Enzyme.configure({ adapter: new Adapter() });