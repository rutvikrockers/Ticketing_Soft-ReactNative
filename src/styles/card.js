import Dimensions from 'Dimensions';
const {width, height} = Dimensions.get('window');
import colors from './colors';
//import { Transform } from 'stream';

const card = {
  overlay: {
    position: 'absolute',
    left: 0,
    top: 0,
    width: width,
    height: height,
    flex: 1
  },
  cardTitle: {
  },
  cardActions: {
    padding: 20,
    flexDirection: 'column',
    justifyContent: 'space-around',
    alignItems: 'stretch',
    alignSelf: 'flex-end',
    backgroundColor: 'rgba(0,0,0,0.25)',
    width: '100%',
    position: 'absolute',
    paddingBottom: 50,
    bottom: 0,
    left: 0
  },
  horizontalGroup: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginTop: 10
  },
  verticalGroup: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    marginTop: 5
  },
  cardTitle: {
    color: '#FFF',
    width: 200,
    fontWeight: 'bold',
    fontSize: 20
  },
  cardCategory: {
    color: '#FFF',
    fontWeight: 'bold',
    fontSize: 14,
    textAlign: 'left'
  }
}

export default card;

