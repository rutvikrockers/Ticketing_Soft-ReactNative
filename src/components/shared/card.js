import React, { Component } from 'react';
import {doFetch} from '../../actions/fetch';
import {
  Text,
  TextInput,
  TouchableOpacity,
  View,
  AsyncStorage
} from 'react-native';
import {connect} from 'react-redux';
import Video from 'react-native-video';
import Dimensions from 'Dimensions';
import {playVideo, isVideoPlaying, doVideoAction} from '../../actions/video';
import Button from '../shared/button';
import Icon from 'react-native-vector-icons/dist/MaterialIcons';
const {width, height} = Dimensions.get('window');
import styles from '../../styles/main.js';
import {onFollow} from '../../actions/follow';
import {Redirect, Link} from 'react-router-native';

class Card extends Component {
  constructor(props) {
    super(props);
    this.player = [];
    this.togglePlay = this.togglePlay.bind(this);
    this.doVideoAction = this.doVideoAction.bind(this);
    this.company_id = ''
    this.token = ''
    this.state = {
      isHidden: false,
    };
  }
  
  async componentWillMount() {
    let mytokens = await AsyncStorage.getItem("token");
    this.token = mytokens;
  }
 
  togglePlay() {
    if(this.props.isPlaying){
      this.props.dispatch(isVideoPlaying(false));
    }else{
      this.props.dispatch(isVideoPlaying(true));
    }
  }
  onBtnFollow() {
    this.props.dispatch(onFollow(this.token,this.company_id))
     ( <Redirect to="../user/followers.js" /> )
  }

  doVideoAction(action, id, value) {
    this.showFeedback(action);
    this.props.dispatch(doVideoAction(action, id, value, this.props.user.token));
  }
  render() {
    let item = this.props.item;
    let index = this.props.index;
    return( 
      <View style={{flex: 1, width: width, height: height, margin: 0, padding: 0}}>
        <Video 
            source={{uri: this.props.uri}}
            ref={(ref) => {
              this.player[item.id] = ref
              this.company_id = item.company_id 
            }}
            resizeMode="cover"
            paused={ (this.props.currentVideo === index && this.props.isPlaying) ? false : true }
            style={styles.backgroundVideo}
            />

        <TouchableOpacity style={styles.overlay} onPress={this.togglePlay}>
        
          <View style={styles.cardActions}>

              <View style={styles.horizontalGroup}>

                  <View style={styles.verticalGroup}>
                    <Text style={styles.cardCategory}>{item.company_category_name}</Text>
                    <Text style={styles.cardTitle}>{item.company_name}</Text>
                  </View>
                  
                  <View style={styles.btnWrapper} key = {item.company_id}>
                    <TouchableOpacity
                      style={styles.btnFollow}
                      onPress={() => {this.onBtnFollow()}}>
                    <Text style={styles.buttonText}>Follow</Text>
                    </TouchableOpacity>
                  </View>
              </View>
                
              <View style={styles.horizontalGroup}>

                <View style={styles.btnWrapper}>
                  <TouchableOpacity
                    onPress={this.props.onPress} style={styles.btnContainer} >
                    <Icon size={25} name="close" color="white" style={styles.footerBtn} />
                  </TouchableOpacity>
                </View>

                <View style={styles.btnWrapper}>
                  <TouchableOpacity
                    onPress={this.props.onPress} style={styles.btnContainer} >
                    <Icon size={25} name="favorite-border" color="white" style={styles.footerBtn} />
                  </TouchableOpacity>
                </View>

              </View>

          </View>
        </TouchableOpacity>
        
      </View>
    );
  }
}

export const mapStateToProps = (state) => {
  return {
    user: state.user,
    feed: state.feed.allVideos,
    path: state.feed.videoPath,
    currentVideo: state.video.nowPlaying,
    isPlaying: state.video.isPlaying
  }
}

export default connect(mapStateToProps)(Card);
