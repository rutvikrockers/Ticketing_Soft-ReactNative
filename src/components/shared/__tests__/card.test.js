import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { mockStore } from '../../../../config/setup';
import toJson from 'enzyme-to-json';
import { default as Card } from '../card';

const render = (props) => shallow(
    <Provider store={mockStore({})}>
        <Card {...props} />
    </Provider>
);

describe('Card', () => {
    let component;
    beforeEach(() => {
        component = render({});
    });

    it('matches its snapshot', () => {
        expect(toJson(component, { mode: 'deep' })).toMatchSnapshot();
    });

    it('renders', () => {
        expect(component.find(Card).length).toBe(1);
    });
});