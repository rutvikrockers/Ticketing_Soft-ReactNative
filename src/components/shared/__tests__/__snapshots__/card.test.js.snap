// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches its snapshot 1`] = `<Connect(Card) />`;
